pkg/config/types/v1alpha1/generate/options.go

```diff
@@ -43,12 +45,22 @@ func WithInstallImage(imageRef string) GenOption {
 	}
 }
 
+// WithNetworkConfig allows to pass network config to be used.
+func WithNetworkConfig(network *v1alpha1.NetworkConfig) GenOption {
```

Comment:
I don't like passing whole `NetworkConfig`, but I don't know better way? pass individual fields? `machine.Device`?

Reply:
This works for now I think. We can figure out a better way in the future IMO.